tests/test_frame.py

```diff
@@ -1386,6 +1386,16 @@ def baz(val: Styler) -> str:
 
     check(assert_type(pd.DataFrame({"a": [1], "b": [1]}).style.pipe(baz), str), str)
 
+    def qux(df: pd.DataFrame, a: int, b: str) -> pd.DataFrame:
+        return pd.DataFrame(df)
+
+    check(
+        assert_type(
+            pd.DataFrame({"a": [1], "b": [1]}).pipe(qux, 1, y="a"), pd.DataFrame
```

Comment:
> Isn't the problem here that y is not an argument of qux?

Embarrassing. Yeah this might be it—forgot to commit my entire diff. Thanks!

> a function with a "normal" argument, a keyword-only argument, and a positional-only argument

Agree on all of these. Didn't think of including the positional only argument, good call there.

> an invalid call by providing an unexpected keyword argument (will need ignores and be inside TYPE_CHECKING_INVALID_USAGE)

Yeah plus also feeding in the wrong type as well to positional, positional-only, keyword, etc.

Reply:
> It might be nice to have at least two tests:
> 
> * a function with a "normal" argument, a keyword-only argument, and a positional-only argument
> * an invalid call by providing an unexpected keyword argument (will need ignores and be inside `TYPE_CHECKING_INVALID_USAGE`)

I agree with @twoertwein  here.  Should also test the `tuple` kind of argument as well.

Reply:
> Should also test the tuple kind of argument as well.

Oh yeah, kinda assumed there were tests already for that since I didn't touch that version of the typing, but I'll look and if they don't already exists—I'll add them